File: app/src/main/java/com/my/televip/virtuals/tgnet/ConnectionsManager.java
package com.my.televip.virtuals.tgnet;

import com.my.televip.Class.ClassLoad;
import com.my.televip.Class.ClassNames;
import com.my.televip.obfuscate.Obfuscate;
import com.my.televip.virtuals.messenger.UserConfig;

import de.robv.android.xposed.XposedHelpers;

public class ConnectionsManager {
    final Object connectionsManager;

    public ConnectionsManager(Object instance)
    {
        this.connectionsManager = instance;
    }

    public void sendRequest(Object object, Object completionBlock) {
        XposedHelpers.callMethod(connectionsManager, Obfuscate.getMethodName("ConnectionsManager", "sendRequest"), object, completionBlock);
    }

    public int getCurrentTime(){
        return (int)XposedHelpers.callMethod(connectionsManager, Obfuscate.getMethodName("ConnectionsManager", "getCurrentTime"));
    }

    public static ConnectionsManager getInstance(int num){
        return new ConnectionsManager(XposedHelpers.callStaticMethod(ClassLoad.getClass(ClassNames.CONNECTIONS_MANAGER), Obfuscate.getMethodName("ConnectionsManager", "getInstance"), num));
    }

    public static ConnectionsManager getConnectionsManager() {
        return getInstance(UserConfig.getSelectedAccount());
    }

}

File: app/src/main/java/com/my/televip/virtuals/tgnet/ConnectionsManager.kt
package com.my.televip.virtuals.tgnet

import com.my.televip.clazz.ClassLoad
import com.my.televip.clazz.ClassNames
import com.my.televip.obfuscate.Obfuscate
import com.my.televip.virtuals.messenger.UserConfig
import de.robv.android.xposed.XposedHelpers

class ConnectionsManager(@JvmField val connectionsManager: Any?) {

    fun sendRequest(obj: Any?, completionBlock: Any?) {
        val target = connectionsManager ?: return
        XposedHelpers.callMethod(target, Obfuscate.getMethodName("ConnectionsManager", "sendRequest"), obj, completionBlock)
    }

    fun getCurrentTime(): Int {
        val target = connectionsManager ?: return 0
        return XposedHelpers.callMethod(target, Obfuscate.getMethodName("ConnectionsManager", "getCurrentTime")) as? Int ?: 0
    }

    companion object {
        @JvmStatic
        fun getInstance(num: Int): ConnectionsManager {
            val cmClass = ClassLoad.getClass(ClassNames.CONNECTIONS_MANAGER)
            val instance = XposedHelpers.callStaticMethod(cmClass, Obfuscate.getMethodName("ConnectionsManager", "getInstance"), num)
            return ConnectionsManager(instance)
        }

        @JvmStatic
        fun getConnectionsManager(): ConnectionsManager {
            return getInstance(UserConfig.getSelectedAccount())
        }
    }
}
